Tiltfile

```diff
@@ -22,18 +23,22 @@ settings = {
     "capi_version": "v1.6.0",
     "cert_manager_version": "v1.13.2",
     "kubernetes_version": "v1.28.0",
-    "aks_kubernetes_version": "v1.26.3",
+    "aks_kubernetes_version": "v1.27.3",
```

Comment:
I think v1.28.3 is available in AKS for all the regions we test. And it might not hurt to bump the plain `kubernetes_version` to the latest v1.28 patch.

Reply:
I am happy to bump it!
I can't seem to find the historical reason for having an older k8s version for AKS. @CecileRobertMichon , @mboersma , was AKS K8s version chosen to an older k8s version (for example: self managed at `1.27` and AKS at `1.25`) only because a lower K8s version was more likely to be present in all azure locations?

Reply:
I think historically the self-managed version has been able to move faster than the AKS one because we can usually build reference images faster than AKS can GA new Kubernetes versions. I don't think we've been holding the AKS version back for any other reason.

Ideally I think we strive to test either the latest version we can or whatever most of the go.mod k8s.io dependencies happen to be, I don't remember. Either way, I think the latest v1.28 patch for each matches that currently.

Reply:
> I think historically the self-managed version has been able to move faster than the AKS one because we can usually build reference images faster than AKS can GA new Kubernetes versions.

+1, that's also my impression. I don't see any reason not to use the latest generally available for AKS though:

```shell
% az aks get-versions 
KubernetesVersion    Upgrades
-------------------  ------------------------
1.28.3               None available
1.28.0               1.28.3
1.27.7               1.28.0, 1.28.3
1.27.3               1.27.7, 1.28.0, 1.28.3
1.26.10              1.27.3, 1.27.7
1.26.6               1.26.10, 1.27.3, 1.27.7
1.25.15              1.26.6, 1.26.10
1.25.11              1.25.15, 1.26.6, 1.26.10
```

Reply:
Thanks for the input everyone! I updated the K8s version to `v1.28.3`.

Reply:
I think there were some problems with Fleet and `1.28.x`. Either that or ClusterClass, I can't remember off the top of my head. I think this shouldn't block the version from being defaulted to `1.28.x`, but I will get back to you on the specifics.